oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -94,6 +95,11 @@ public void refresh() throws IOException {
       requestMetadata = null;
       temporaryAccess = null;
       temporaryAccess = refreshAccessToken();
+      if (changeListeners != null) {
+        for (CredentialsChangedListener listener : changeListeners) {
```

Comment:
This is happening under synchronized(lock) earlier in the method. This is similar to the V1 analog:
https://github.com/google/google-oauth-java-client/blob/dev/google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/Credential.java#L485